src/dicom-data.c

```diff
@@ -2403,7 +2404,9 @@ void dcm_frame_destroy(DcmFrame *frame)
 
 // Basic Offset Table
 
-DcmBOT *dcm_bot_create(DcmError **error, ssize_t *offsets, uint32_t num_frames)
+DcmBOT *dcm_bot_create(DcmError **error, 
+                       ssize_t *offsets, uint32_t num_frames, 
+                       ssize_t first_frame_offset)
```

Comment:
We may ultimately want to use `uint64_t` instead of `ssize_t` (see https://github.com/ImagingDataCommons/libdicom/issues/21)

Reply:
I agree, there are a load of types that need changing, I think. I put some notes in the TODO. Let's go over them all in another PR.